signer/src/storage/postgres.rs

```diff
@@ -735,6 +735,38 @@ impl super::DbRead for PgStore {
         .map_err(Error::SqlxQuery)
     }
 
+    async fn can_sign_deposit_tx(
```

Comment:
I'm not suggesting we change this here and now, but just wanted to note that this is an example of where I probably would have split this up into 1) getting the valid keys, 2) getting the deposit request, and 3) performing the validation in Rust.

I think that would have let you perform more of the testing as simpler unit tests.

Reply:
Very good point

Reply:
🙏🏽